crates/types/src/config/worker.rs

```diff
@@ -287,6 +287,7 @@ impl StorageOptions {
     pub fn apply_common(&mut self, common: &CommonOptions) {
         self.rocksdb.apply_common(&common.rocksdb);
         if self.num_partitions_to_share_memory_budget.is_none() {
+            // todo(azmy): this value needs to eventually come from cluster configuration seed.
```

Comment:
This is a good todo to resolve actually. Otherwise our memory management might be at risk a bit.

Reply:
I was trying to figure this out for a while and it's not very easy to fix. The problem this value is used in early boot stages to create the PartitionStoreManager and by the time the partition table is provisioned (externally) it would be too late. 

I am trying to figure out if those options can be manipulated later and then we can change it on provision. 

A much simpler solution is to default no `number of cpus` if not explicitly set by the user.